Source/CrossPlatformArcadiaWatcher.cs

```diff
@@ -0,0 +1,109 @@
+using System.IO;
+using System.Collections.Generic;
+using System.Threading;
+using clojure.lang;
+using Godot;
+
+public class CrossPlatformArcadiaWatcher
+{
+    // Polls the FS
+    private System.Threading.Timer poll_timer;
+
+    // Reloads files
+    private System.Threading.Timer drain_timer;
+
+    private Dictionary<string, bool> changed_files = new Dictionary<string, bool>();
+    private Dictionary<string, FileInfo> last_files;
+
+    private bool verbose_logging = false;
+
+    public CrossPlatformArcadiaWatcher(bool verbose)
+    {
+        this.verbose_logging = verbose;
+        poll_timer = new System.Threading.Timer(CheckStatus, new AutoResetEvent(false), 100, 100);
+        drain_timer = new System.Threading.Timer(DrainChanges, new AutoResetEvent(false), 100, 100);
```

Comment:
I'm not sure what sane numbers are here, having these match obviously make having _two_ timers pointless but I wanted to maintain the ability to scale them separately for performance reasons.